### PR TITLE
chore: finalize security setup

### LINKED ISSUE #181: Public Repo - Security Final touches
Once repository is public, activate the security settings that cannot be activate in private state i.e., 

- Activate CodeQL
- Enable private vulnerability reporting ([docu](https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository)) and adjust README.md.

### PR DESCRIPTION
**Pull request by lechnerc77**
## Purpose

* Adds CodeQL Setup for Go
* Adjusts README for private vulnerability reporting

closes #181 

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
Execute CodeQL GitHub Action
```

## What to Check

Verify that the following are valid

* Checks get executed

## Other Information

More info on private vulnerability reporting: <https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ The [developer documentation](DEVELOPER.md) file is a basic outline on how to bu
 
 For more information about how to contribute, the project structure, and additional contribution information, see our [Contribution Guidelines](CONTRIBUTING.md).
 
-> **Note**: We take Terraform's security and our users' trust very seriously. If you believe you have found a security issue in the Terraform provider for SAP BTP, please responsibly disclose it by using the *private reporting of vulnerabilities* in this repository.
+> **Note**: We take Terraform's security and our users' trust very seriously. If you believe you have found a security issue in the Terraform provider for SAP BTP, please responsibly disclose it. You find more details on the process in the [SECURITY.md] file.
```
Comment (Shegox) [inline README.md]:
```suggestion
> **Note**: We take Terraform's security and our users' trust very seriously. If you believe you have found a security issue in the Terraform provider for SAP BTP, please responsibly disclose it. You find more details on the process in [our security policy](https://github.com/SAP/terraform-provider-btp/security/policy).
```

File: SECURITY.md
Comment (Shegox) [inline SECURITY.md:1]: if you don't derive from the default SAP Security Policy, you don't need an own SECURITY.md and GitHub will use the [org default](https://github.com/SAP/.github/blob/main/SECURITY.md) for https://github.com/SAP/terraform-provider-btp/security/policy (which allows us to change it centrally without changing it in each repository) 

--> you can delete the SECURITY.md

Comment (lechnerc77) [reply SECURITY.md:1]: Removed

**Review by Shegox — COMMENTED** [review]
Changes about the README/Security Policy look good from my side 👍

**Review by v0lkc — APPROVED** [review]

**Pull request merged**